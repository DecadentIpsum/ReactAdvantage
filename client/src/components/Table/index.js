import React, { Component } from 'react';
import { DataTable } from 'primereact/components/datatable/DataTable';

import Row from './components/Row';
import Column from './components/Column';

export default function Table({ tableRef, ...props }) {
  return (
    <DataTable {...props} ref={tableRef}>
      {props.children}
    </DataTable>
  );
}

export { Row, Column };
