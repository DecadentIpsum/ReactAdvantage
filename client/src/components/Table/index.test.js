import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Table from './index';

it('renders correctly', () => {
    const component = shallow(<Table />);

    expect(component).toMatchSnapshot();
});