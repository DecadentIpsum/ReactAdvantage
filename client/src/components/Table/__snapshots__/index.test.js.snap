// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alwaysShowPaginator": true,
      "autoLayout": false,
      "children": undefined,
      "className": null,
      "columnResizeMode": "fit",
      "compareSelectionBy": "deepEquals",
      "contextMenu": null,
      "csvSeparator": ",",
      "dataKey": null,
      "emptyMessage": "No records found",
      "expandedRows": null,
      "exportFilename": "download",
      "filters": null,
      "first": null,
      "footer": null,
      "footerColumnGroup": null,
      "frozenFooterColumnGroup": null,
      "frozenHeaderColumnGroup": null,
      "frozenValue": null,
      "frozenWidth": null,
      "globalFilter": null,
      "header": null,
      "headerColumnGroup": null,
      "id": null,
      "lazy": false,
      "loading": false,
      "loadingIcon": "fa-circle-o-notch",
      "metaKeySelection": true,
      "multiSortMeta": null,
      "onColReorder": null,
      "onColumnResizeEnd": null,
      "onContextMenuSelect": null,
      "onFilter": null,
      "onLazyLoad": null,
      "onPage": null,
      "onRowClick": null,
      "onRowCollapse": null,
      "onRowDoubleClick": null,
      "onRowExpand": null,
      "onRowReorder": null,
      "onRowSelect": null,
      "onRowToggle": null,
      "onRowUnselect": null,
      "onSelectionChange": null,
      "onSort": null,
      "pageLinkSize": 5,
      "paginator": false,
      "paginatorLeft": null,
      "paginatorPosition": "bottom",
      "paginatorRight": null,
      "paginatorTemplate": "FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink RowsPerPageDropdown",
      "reorderableColumns": false,
      "resizableColumns": false,
      "responsive": false,
      "rowClassName": null,
      "rowExpansionTemplate": null,
      "rowGroupFooterTemplate": null,
      "rowGroupHeaderTemplate": null,
      "rowGroupMode": null,
      "rows": null,
      "rowsPerPageOptions": null,
      "scrollHeight": null,
      "scrollable": false,
      "selection": null,
      "selectionMode": null,
      "sortField": null,
      "sortMode": "single",
      "sortOrder": null,
      "style": null,
      "tableClassName": null,
      "tableStyle": null,
      "totalRecords": null,
      "unfrozenWidth": null,
      "value": null,
      "virtualScroll": false,
      "virtualScrollDelay": 500,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alwaysShowPaginator": true,
        "autoLayout": false,
        "children": undefined,
        "className": null,
        "columnResizeMode": "fit",
        "compareSelectionBy": "deepEquals",
        "contextMenu": null,
        "csvSeparator": ",",
        "dataKey": null,
        "emptyMessage": "No records found",
        "expandedRows": null,
        "exportFilename": "download",
        "filters": null,
        "first": null,
        "footer": null,
        "footerColumnGroup": null,
        "frozenFooterColumnGroup": null,
        "frozenHeaderColumnGroup": null,
        "frozenValue": null,
        "frozenWidth": null,
        "globalFilter": null,
        "header": null,
        "headerColumnGroup": null,
        "id": null,
        "lazy": false,
        "loading": false,
        "loadingIcon": "fa-circle-o-notch",
        "metaKeySelection": true,
        "multiSortMeta": null,
        "onColReorder": null,
        "onColumnResizeEnd": null,
        "onContextMenuSelect": null,
        "onFilter": null,
        "onLazyLoad": null,
        "onPage": null,
        "onRowClick": null,
        "onRowCollapse": null,
        "onRowDoubleClick": null,
        "onRowExpand": null,
        "onRowReorder": null,
        "onRowSelect": null,
        "onRowToggle": null,
        "onRowUnselect": null,
        "onSelectionChange": null,
        "onSort": null,
        "pageLinkSize": 5,
        "paginator": false,
        "paginatorLeft": null,
        "paginatorPosition": "bottom",
        "paginatorRight": null,
        "paginatorTemplate": "FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink RowsPerPageDropdown",
        "reorderableColumns": false,
        "resizableColumns": false,
        "responsive": false,
        "rowClassName": null,
        "rowExpansionTemplate": null,
        "rowGroupFooterTemplate": null,
        "rowGroupHeaderTemplate": null,
        "rowGroupMode": null,
        "rows": null,
        "rowsPerPageOptions": null,
        "scrollHeight": null,
        "scrollable": false,
        "selection": null,
        "selectionMode": null,
        "sortField": null,
        "sortMode": "single",
        "sortOrder": null,
        "style": null,
        "tableClassName": null,
        "tableStyle": null,
        "totalRecords": null,
        "unfrozenWidth": null,
        "value": null,
        "virtualScroll": false,
        "virtualScrollDelay": 500,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
